mlir/lib/Conversion/TosaToRock/TosaToRock.cpp

```diff
@@ -140,21 +140,27 @@ getArchAttributes(Operation *op, Type inputType) {
 }
 
 static FailureOr<rock::ConvOp>
-makeRockConv2D(ConversionPatternRewriter &rw, Operation *op, Value input,
-               Value filter, Value output, const DenseI64ArrayAttr &pad,
-               const DenseI64ArrayAttr &stride,
-               const DenseI64ArrayAttr &dilation, int64_t group) {
+makeRockConv(ConversionPatternRewriter &rw, Operation *op, Value input,
+             Value filter, Value output, const DenseI64ArrayAttr &pad,
+             const DenseI64ArrayAttr &stride, const DenseI64ArrayAttr &dilation,
+             int64_t group) {
   Location loc = op->getLoc();
 
-  SmallString<5> filterLayout("kyxc");
+  SmallString<7> filterLayout("kyxc");
```

Comment:
On further study, I can't make these SmallStrings into MutableArrayRef in expandTensor(), because part of the expansion is to insert a "g" into the layout string.  As a MutableArrayRef, I can modify the existing elements, but since the buffer is owned by somebody else, I can't expand it.  Therefore, SmallString<8> all around.